apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -264,7 +340,13 @@ export const pageRouter = router({
 
   getRootPage: protectedProcedure
     .input(getRootPageSchema)
-    .query(async ({ input: { siteId } }) => {
+    .query(async ({ ctx, input: { siteId } }) => {
+      await validateUserPermissions({
+        userId: ctx.user.id,
+        siteId,
```

Comment:
Should we inject siteid into the session context when users login and pick it up from there?

Reply:
we _can't_ do this at the point of login because a user can have access to >1 site, which is also why the query itself requires a `siteId`.